src/Microsoft.AspNetCore.Rewrite/Internal/UrlActions/CustomResponseAction.cs

```diff
@@ -0,0 +1,51 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Text;
+using Microsoft.AspNetCore.Http.Extensions;
+using Microsoft.AspNetCore.Http.Features;
+using Microsoft.AspNetCore.Rewrite.Logging;
+
+namespace Microsoft.AspNetCore.Rewrite.Internal.UrlActions
+{
+    public class CustomResponseAction : UrlAction
+    {
+        public int StatusCode { get; }
+        public string StatusReason { get; }
+        public string StatusDescription { get; }
+
+        public CustomResponseAction(int statusCode, string statusReason, string statusDescription)
+        {
+            if (string.IsNullOrEmpty(statusReason))
```

Comment:
are statusReason and statusDescription really required? A response only requires a status code.